engine/src/main/java/io/zeebe/engine/processing/streamprocessor/writers/StateWriter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processing.streamprocessor.writers;
+
+import io.zeebe.engine.state.EventAppliers;
+import io.zeebe.msgpack.UnpackedObject;
+import io.zeebe.protocol.impl.record.RecordMetadata;
+import io.zeebe.protocol.record.intent.Intent;
+import java.util.function.Consumer;
+
+/**
+ * This event writer changes state for the events it writes to the stream.
+ *
+ * <p>Note that it does not write events to the stream itself, but it delegates this to the {@link
+ * TypedStreamWriter}.
+ *
+ * <p>Note that it does not change the state itself, but delegates this to the {@link
+ * EventAppliers}.
+ */
+public final class StateWriter implements TypedEventWriter {
+
+  private static final Consumer<RecordMetadata> NO_METADATA = metadata -> {};
```

Comment:
I don't understand the name and the concept in general. As a name I would propose ´IGNORE_METADATA` because this describes the behavior of the consumer. 

Still I would appreciate it we could have a short chat about the idea behind these consumers.

Reply:
In other places this is called `NOOP`, since it does nothing with the record metadata that is returned. I think that fits here as well.